### ISSUE TITLE
Using weave with podman

### ISSUE THREAD

**Issue by giminni** [weaveworks/weave#3755]
Is there a chance to use weave CNI plugin with podman?

I tried a couple to start weaveexec with `sudo podman run --rm --privileged --net host --pid host -v /:/host -e HOST_ROOT=/host -e DOCKERHUB_USER=weaveworks -e WEAVE_VERSION -e WEAVE_DEBUG -e WEAVE_DOCKER_ARGS -e WEAVE_PASSWORD -e WEAVE_PORT -e WEAVE_HTTP_ADDR -e WEAVE_STATUS_ADDR -e WEAVE_CONTAINER_NAME -e WEAVE_MTU -e WEAVE_NO_FASTDP -e WEAVE_NO_BRIDGED_FASTDP -e DOCKER_BRIDGE -e DOCKER_CLIENT_HOST= -e DOCKER_CLIENT_ARGS -e PROXY_HOST=127.0.0.1 -e COVERAGE -e CHECKPOINT_DISABLE -e AWSVPC weaveworks/weaveexec:2.5.2 --local launch`

Got `Failed to get netdev for "docker0" bridge: Link not found`

Should I define a custom bridge called `docker0`?

**Comment by primeroz** [comment]
Having exactly the same issue with 2.6 
@giminni  Did you get anywhere with this ?

**Comment by primeroz** [comment]
even creating a fake "docker0" bridge does not help

it does look like the weave helper script is really geared only towards docker ( Probably for the proxy feature ) 

```
 podman run --rm --privileged --net host --pid host -v /:/host -e HOST_ROOT=/host -e DOCKERHUB_USER=weaveworks -e WEAVE_VERSION -e WEAVE_DEBUG -e WEAVE_DOCKER_ARGS -e WEAVE_PASSWORD -e WEAVE_PORT -e WEAVE_HTTP_ADDR -e WEAVE_STATUS_ADDR -e WEAVE_CONTAINER_NAME -e WEAVE_MTU -e WEAVE_NO_FASTDP -e WEAVE_NO_BRIDGED_FASTDP  -e DOCKER_CLIENT_HOST= -e DOCKER_CLIENT_ARGS -e PROXY_HOST=127.0.0.1 -e COVERAGE -e CHECKPOINT_DISABLE -e AWSVPC -e WEAVE_DEBUG weaveworks/weaveexec:2.6.0 --local launch archpi  
cannot locate running docker daemon
Warning: unable to detect proxy TLS configuration. To enable TLS, launch the proxy with 'weave launch' and supply TLS options. To suppress this warning, supply the '--no-detect-tls' option.
unable to create container: Get http://unix.sock/v1.18/version: dial unix /var/run/docker.sock: connect: no such file or directory
```

**Comment by bboreham** [comment]
Yes, the `weave` script assumes Docker.  If you have necessary tools installed you can run `weave --local`.

Add `--docker-api=''` to stop it trying to talk to Docker. Take a look at how the [Kubernetes Daemonset image](https://github.com/weaveworks/weave/tree/master/prog/weave-kube) runs - it has no contact with Docker.

Happy to review PRs to make it work better with Podman.

**Comment by primeroz** [comment]
Thanks for the pointer , i will give it a shot and see if i can come up with changes that will support podman to the script 

@bboreham in general do you prefer 
* a flag to the script like `--podman` 
* a try an error in the script to detect if `docker` is present else failover to `podman` 
* a third option i did not think about 

?

**Comment by bboreham** [comment]
I know almost nothing about Podman.  I guess `--podman` would parallel `--local`.
If it has an API like Docker then `--podman-api` would parallel `--docker-api`.
Perhaps an environment variable.  Probably not magic failover.
Sadly the script has hundreds of odd corners - it's ok to get just a couple of main tasks working.

**Comment by fpoirotte** [comment]
podman has a docker-compatible API socket (since podman v4 I believe) which can be activated through systemd with `systemctl start podman` (which simply runs `podman system service` in the background for you)

The actual socket's location will vary depending on how you use podman (rootful or rootless).
The following command (tested with podman v4.3.1) will give you its location: `podman info -f json | jq -r .host.remoteSocket.path`

Therefore, you should be able to go past the error above (`Unable to start docker client: Get http://unix.sock/version: dial unix /var/run/docker.sock: connect: no such file or directory`) by mounting podman's API socket at /var/run/docker.sock inside the container.

In your commands, you may also need to adapt `--docker-bridge docker0` so as to match podman's default bridge network (`podman`)

**Comment by byrnedo** [comment]
@primeroz did you get anywhere with supporting podman?

**Comment by lchunleo** [comment]
May I check if podman can be used with weavenet ?
